# HG changeset patch
# User Rik <rik@octave.org>
# Date 1363024184 25200
#      Mon Mar 11 10:49:44 2013 -0700
# Node ID 98d8d1f4b7f8b4281909876189803d0850f73f45
# Parent  b1b01c69967e77bdf7507be7f5ad5d958c90ffe8
build: Add creation of Doxygen docs to build system

* configure.ac: Add doc/doxyhtml/Makefile to list of Makefiles to generate.

* Doxyfile: Moved and renamed to doc/doxyhtml/Doxygen.cfg.

* doc/Makefile.am: Add doxyhtml directory to list of SUBDIRS.

* doc/doxyhtml/Doxygen.cfg: Renamed from Doxyfile at top-level.

* doc/doxyhtml/Makefile.am: New Makefile.am with rules for building
and cleaning Doxygen documentation

* doc/doxyhtml/README: README file explaining how to build optional
Doxygen documentation.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2852,16 +2852,17 @@ AC_OUTPUT_MAKE_DEFS
 ### Do the substitutions in all the Makefiles.
 
 AC_SUBST(ac_config_files)
 AC_SUBST(ac_config_headers)
 
 AC_CONFIG_FILES([
   Makefile 
   doc/Makefile
+  doc/doxyhtml/Makefile
   doc/icons/Makefile
   doc/interpreter/Makefile
   doc/liboctave/Makefile
   doc/refcard/Makefile
   examples/Makefile
   libgnu/Makefile
   libgui/Makefile
   libinterp/Makefile
diff --git a/doc/Makefile.am b/doc/Makefile.am
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -24,10 +24,10 @@ include $(top_srcdir)/build-aux/common.m
 ## which can lead to a confusing error message stream.
 .NOTPARALLEL:
 
 EXTRA_DIST = \
   Makefile.am \
   texinfo.tex \
   texmf.cnf
 
-SUBDIRS = icons interpreter liboctave refcard
+SUBDIRS = doxyhtml icons interpreter liboctave refcard
 
diff --git a/Doxyfile b/doc/doxyhtml/Doxygen.cfg
rename from Doxyfile
rename to doc/doxyhtml/Doxygen.cfg
diff --git a/doc/doxyhtml/Makefile.am b/doc/doxyhtml/Makefile.am
new file mode 100644
--- /dev/null
+++ b/doc/doxyhtml/Makefile.am
@@ -0,0 +1,32 @@
+# Makefile for Octave's doc/doxyhtml directory
+#
+# Copyright (C) 1993-2012 John W. Eaton
+#
+# This file is part of Octave.
+# 
+# Octave is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at
+# your option) any later version.
+# 
+# Octave is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with Octave; see the file COPYING.  If not, see
+# <http://www.gnu.org/licenses/>.
+
+include $(top_srcdir)/build-aux/common.mk
+
+doxyhtml:
+	cd ../..; doxygen doc/doxyhtml/Doxygen.cfg 
+
+EXTRA_DIST = \
+  Doxygen.cfg \
+  Makefile.am \
+  README
+
+maintainer-clean-local:
+	rm -rf `ls | $(GREP) -v Doxygen.cfg | $(GREP) -v Makefile.am | $(GREP) -v Makefile.in | $(GREP) -v README`
diff --git a/doc/doxyhtml/README b/doc/doxyhtml/README
new file mode 100644
--- /dev/null
+++ b/doc/doxyhtml/README
@@ -0,0 +1,11 @@
+This directory contains documentation in Doxygen format for
+Octave's source code.  It is not created by default.
+
+To produce Doxygen documentation use
+
+make doxyhtml
+
+Doxygen documentation can be helpful for developers of Octave, but is not
+needed by users of Octave.  In addition, the documentation requires
+approximately 1.2GB of storage space.  For these reasons it is not maintained
+under version control nor distributed in tarballs.
